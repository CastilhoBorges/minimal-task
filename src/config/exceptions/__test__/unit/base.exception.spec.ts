import { HttpStatus } from '@nestjs/common';
import { BaseException } from '../../base.exception';

describe('BaseException', () => {
  it('should create an instance with required parameters', () => {
    const exception = new BaseException({
      name: 'TestError',
      message: 'This is a test error',
      statusCode: HttpStatus.BAD_REQUEST,
    });

    expect(exception).toBeInstanceOf(BaseException);
    expect(exception.getResponse()).toEqual({
      name: 'TestError',
      message: 'This is a test error',
      statusCode: HttpStatus.BAD_REQUEST,
      metadata: undefined,
      stack: undefined,
    });
    expect(exception.getStatus()).toBe(HttpStatus.BAD_REQUEST);
    expect(exception.getResponse()).toMatchSnapshot();
  });

  it('should create an instance with stack trace', () => {
    const stackTrace = 'Error: Something went wrong\n at someFile.ts:10:5';
    const exception = new BaseException({
      name: 'TestError',
      message: 'This is a test error',
      statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
      metadata: undefined,
      stack: stackTrace,
    });

    expect(exception.getResponse()).toEqual({
      name: 'TestError',
      message: 'This is a test error',
      statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
      metadata: undefined,
      stack: stackTrace,
    });
    expect(exception.getResponse()).toMatchSnapshot();
  });

  it('should create an instance with metadata', () => {
    const metadata = { key: 'value', errorCode: 1234 };
    const exception = new BaseException({
      name: 'TestError',
      message: 'This is a test error',
      statusCode: HttpStatus.NOT_FOUND,
      metadata: metadata,
      stack: undefined,
    });

    expect(exception.getResponse()).toEqual({
      name: 'TestError',
      message: 'This is a test error',
      statusCode: HttpStatus.NOT_FOUND,
      metadata,
      stack: undefined,
    });
    expect(exception.getResponse()).toMatchSnapshot();
  });

  it('should create an instance with all parameters', () => {
    const stackTrace = 'Error: Something went wrong\n at someFile.ts:10:5';
    const metadata = { debug: true };

    const exception = new BaseException({
      name: 'ComplexError',
      message: 'This is a complex error',
      statusCode: HttpStatus.FORBIDDEN,
      metadata: metadata,
      stack: stackTrace,
    });

    expect(exception.getResponse()).toEqual({
      name: 'ComplexError',
      message: 'This is a complex error',
      statusCode: HttpStatus.FORBIDDEN,
      metadata,
      stack: stackTrace,
    });
    expect(exception.getResponse()).toMatchSnapshot();
  });
});
