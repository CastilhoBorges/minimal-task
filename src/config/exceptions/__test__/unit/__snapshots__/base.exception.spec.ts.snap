// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseException should create an instance with all parameters 1`] = `
{
  "message": "This is a complex error",
  "metadata": {
    "debug": true,
  },
  "name": "ComplexError",
  "stack": "Error: Something went wrong
 at someFile.ts:10:5",
  "statusCode": 403,
}
`;

exports[`BaseException should create an instance with metadata 1`] = `
{
  "message": "This is a test error",
  "metadata": {
    "errorCode": 1234,
    "key": "value",
  },
  "name": "TestError",
  "stack": undefined,
  "statusCode": 404,
}
`;

exports[`BaseException should create an instance with required parameters 1`] = `
{
  "message": "This is a test error",
  "metadata": undefined,
  "name": "TestError",
  "stack": undefined,
  "statusCode": 400,
}
`;

exports[`BaseException should create an instance with stack trace 1`] = `
{
  "message": "This is a test error",
  "metadata": undefined,
  "name": "TestError",
  "stack": "Error: Something went wrong
 at someFile.ts:10:5",
  "statusCode": 500,
}
`;
